api.js

```diff
@@ -125,21 +139,25 @@ class Api extends EventEmitter {
 						// helpers from within the `resolveTestsFrom` directory. Without
 						// arguments this is the `projectDir`, else it's `process.cwd()`
 						// which may be nested too deeply.
-						return new AvaFiles({cwd: this.options.resolveTestsFrom}).findTestHelpers().then(helpers => {
-							return {
-								cacheDir: precompilation.cacheDir,
-								map: files.concat(helpers).reduce((acc, file) => {
-									try {
-										const realpath = fs.realpathSync(file);
-										const hash = precompilation.precompiler.precompileFile(realpath);
-										acc[realpath] = hash;
-									} catch (err) {
-										throw Object.assign(err, {file});
-									}
-									return acc;
-								}, {})
-							};
-						});
+						return new AvaFiles({cwd: this.options.resolveTestsFrom, extensions: allExtensions})
+							.findTestHelpers().then(helpers => {
+								return {
+									cacheDir: precompilation.cacheDir,
+									map: files.concat(helpers).reduce((acc, file) => {
+										try {
+											const realpath = fs.realpathSync(file);
+											let hash = realpath;
+											if (!doNotCompileExtensions.includes(path.extname(realpath))) {
```

Comment:
OK so the tricky thing is we still want to compile enhancements, unless that's *also* disabled. This currently does use Babel, but it won't apply AVA's regular presets. I think this means instantiating a second `precompilation` object, and doing a second build of Babel configuration.

I feel bad for not spotting this earlier, and it's a bit tricky to line all this up correctly. Let me know if you want me to push a commit for this instead.

Reply:
@novemberborn If you could on this one I can handle the rest, work is a bit busy so I don't have alot of time to work on this. Thanks for all the help!